packages/eas-cli/src/commands/update/index.ts

```diff
@@ -241,14 +242,39 @@ export default class UpdatePublish extends EasCommand {
       const assets = filterExportedPlatformsByFlag(collectedAssets, platformFlag);
       realizedPlatforms = Object.keys(assets) as PublishPlatform[];
 
-      const uploadResults = await uploadAssetsAsync(
-        graphqlClient,
-        assets,
-        projectId,
-        (totalAssets, missingAssets) => {
-          assetSpinner.text = `Uploading (${totalAssets - missingAssets}/${totalAssets})`;
-        }
-      );
+      let lastUploadedStorageKeys = new Set<string>();
+      let lastAssetUploadResults: {
+        asset: RawAsset & { storageKey: string };
+        finished: boolean;
+      }[] = [];
+      let timeAtWhichToTimeout = Date.now() + 60 * 1000; // sixty seconds from now
+
+      const uploadResults = await Promise.race([
+        uploadAssetsAsync(graphqlClient, assets, projectId, assetUploadResults => {
+          const currentUploadedStorageKeys = new Set(
+            assetUploadResults.filter(r => r.finished).map(r => r.asset.storageKey)
+          );
+          if (!setEquals(currentUploadedStorageKeys, lastUploadedStorageKeys)) {
+            timeAtWhichToTimeout = Date.now() + 60 * 1000; // reset timeout to sixty seconds from now
+            lastUploadedStorageKeys = currentUploadedStorageKeys;
+            lastAssetUploadResults = assetUploadResults;
+          }
+
+          const totalAssets = assetUploadResults.length;
+          const missingAssetCount = assetUploadResults.filter(a => !a.finished).length;
+          assetSpinner.text = `Uploading (${totalAssets - missingAssetCount}/${totalAssets})`;
+        }),
+        // time out when we haven't successfully uploaded/processed any assets in the last 60 seconds
+        (async () => {
+          while (Date.now() < timeAtWhichToTimeout) {
+            await new Promise(res => setTimeout(res, 1000)); // wait 1 second
+          }
```

Comment:
Maybe I'm missing sth but why isn't it just:
```
await new Promise(res => setTimeout(res, 60 * 1000));
```

Reply:
Added a comment about how the timeout mechanism works